src/main/kotlin/org/wa/rceditor/application/model/MainModel.kt

```diff
@@ -1,26 +1,310 @@
+import org.wycliffeassociates.resourcecontainer.entity.DublinCore
+import org.wycliffeassociates.resourcecontainer.entity.Manifest
+import tornadofx.*
+import java.io.File
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 import kotlin.system.exitProcess
 
 class MainModel {
 
+    // --------------- Properties ------------------ //
+
+    private var conformsto: String by property()
+    val conformstoProperty = getProperty(MainModel::conformsto)
+
+    private var creator: String by property()
+    val creatorProperty = getProperty(MainModel::creator)
+
+    private var description: String by property()
+    val descriptionProperty = getProperty(MainModel::description)
+
+    private var format: String by property()
+    val formatProperty = getProperty(MainModel::format)
+
+    private var identifier: String by property()
+    val identifierProperty = getProperty(MainModel::identifier)
+
+    private var issued: LocalDate by property()
+    val issuedProperty = getProperty(MainModel::issued)
+
+    private var modified: LocalDate by property()
+    val modifiedProperty = getProperty(MainModel::modified)
+
+    private var languageDirection: String by property()
+    val languageDirectionProperty = getProperty(MainModel::languageDirection)
+
+    private var languageIdentifier: String by property()
+    val languageIdentifierProperty = getProperty(MainModel::languageIdentifier)
+
+    private var languageTitle: String by property()
+    val languageTitleProperty = getProperty(MainModel::languageTitle)
+
+    private var publisher: String by property()
+    val publisherProperty = getProperty(MainModel::publisher)
+
+    private var rights: String by property()
+    val rightsProperty = getProperty(MainModel::rights)
+
+    private var subject: String by property()
+    val subjectProperty = getProperty(MainModel::subject)
+
+    private var title: String by property()
+    val titleProperty = getProperty(MainModel::title)
+
+    private var type: String by property()
+    val typeProperty = getProperty(MainModel::type)
+
+    private var version: String by property()
+    val versionProperty = getProperty(MainModel::version)
+
+    private var checkingLevel: String by property()
+    val checkingLevelProperty = getProperty(MainModel::checkingLevel)
+
+    private val contributors = SortedFilteredList<ContributorItem>()
+    private val relations = SortedFilteredList<RelationItem>()
+    private val sources = SortedFilteredList<SourceItem>()
+    private val checkingEntities = SortedFilteredList<CheckingEntityItem>()
+    private val projects = SortedFilteredList<ProjectItem>()
+
+    private var directoryLoaded: Boolean by property()
+    val directoryLoadedProperty = getProperty(MainModel::directoryLoaded)
+
+    lateinit var container: ResourceContainer
+    lateinit var directory: File
+
+    // -------------- Functions --------------- //
+
+    fun contributors(): ObservableList<ContributorItem> {
```

Comment:
I think this might be better as a property instead of a function

Reply:
@mbr4477 That's a great advice, thanks!